lib/src/parser.dart

```diff
@@ -0,0 +1,222 @@
+    Map<String, dynamic> map = json.decode(jsonString);
+    return createInsertionOrder(map);
+  }
+
+  /// Creates an [InsertionOrder] instance from [map].
+  ///
+  /// Throws an [FormatException] if a required field is missing.
+  static InsertionOrder createInsertionOrder(Map<String, dynamic> map) {
+    final insertionOrder = InsertionOrder();
+
+    if (!map.containsKey('advertiserId')) throw FormatException;
+    insertionOrder.advertiserId = map['advertiserId'];
+
+    if (!map.containsKey('campaignId')) throw FormatException;
+    insertionOrder.campaignId = map['campaignId'];
+
+    if (!map.containsKey('insertionOrderId')) throw FormatException;
+    insertionOrder.insertionOrderId = map['insertionOrderId'];
+
+    if (!map.containsKey('displayName')) throw FormatException;
+    insertionOrder.displayName = map['displayName'];
+
+    if (!map.containsKey('updateTime')) throw FormatException;
+    insertionOrder.updateTime = map['updateTime'];
+
+    if (!map.containsKey('entityStatus')) throw FormatException;
+    insertionOrder.entityStatus = createEntityStatus(map['entityStatus']);
+
+    if (!map.containsKey('pacing')) throw FormatException;
+    insertionOrder.pacing = createPacing(map['pacing']);
+
+    if (!map.containsKey('budget')) throw FormatException;
+    insertionOrder.budget = createBudget(map['budget']);
+    return insertionOrder;
+  }
+
+  /// Creates an [InsertionOrder_Pacing] instance from [map].
+  ///
+  /// Throws an [FormatException] if a required field is missing or if
+  /// both `dailyMaxMicros` and `dailyMaxImpressions` are present.
+  static InsertionOrder_Pacing createPacing(Map<String, dynamic> map) {
+    final pacing = InsertionOrder_Pacing();
+
+    if (!map.containsKey('pacingPeriod')) throw FormatException;
+
+    pacing.pacingPeriod = createPacingPeriod(map['pacingPeriod']);
+
+    if (!map.containsKey('pacingType')) throw FormatException;
+    pacing.pacingType = createPacingType(map['pacingType']);
+
+    if (map.containsKey('dailyMaxMicros') &&
+        map.containsKey('dailyMaxImpressions')) {
+      throw FormatException;
+    } else if (map.containsKey('dailyMaxMicros')) {
+      pacing.dailyMaxMicros = map['dailyMaxMicros'];
+    } else if (map.containsKey('dailyMaxImpressions')) {
+      pacing.dailyMaxImpressions = map['dailyMaxImpressions'];
+    }
+
+    return pacing;
+  }
+
+  /// Creates an [InsertionOrder_InsertionOrderBudget] from [map].
+  ///
+  /// Throws an [FormatException] if a required field is missing.
+  static InsertionOrder_InsertionOrderBudget createBudget(
+      Map<String, dynamic> map) {
+    final budget = InsertionOrder_InsertionOrderBudget();
+
+    if (!map.containsKey('budgetUnit')) throw FormatException;
+    budget.budgetUnit = createBudgetUnit(map['budgetUnit']);
+
+    budget.automationType = createAutomationType(map['automationType']);
+
+    if (!map.containsKey('budgetSegments')) throw FormatException;
+    for (var map in map['budgetSegments']) {
```

Comment:
Is the type in map['budgetSegments'], is a list? If so you can use the map() method:
  budget.budgetSegments = map['budgetSegments'].map(createBudgetSegment).toList();

I think however that the variable name 'map' is pretty confusing, you can instead call it something like jsonData or something like that instead.

Reply:
Yeah, that var name 'map' is confusing and has now been changed.  Unfortunately map() cannot be used for this field, since it is denoted as repeated in the proto definition, and Dart doesn't generate setters for repeated fields. It can only call add(). But I will keep in mind to take advantage of this map() method in other cases that are applicable!